### PR TITLE
[W5][M11-2]Ong Eu Jin, Justin

### PR DESCRIPTION
**Pull request by Justin-Ong**
Added Edit function.
Updated Tests.
Updated User Guide.

### CODE REVIEW

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -79,6 +79,9 @@ public Command parseCommand(String userInput) {
         case DeleteCommand.COMMAND_WORD:
             return prepareDelete(arguments);
 
+        case seedu.addressbook.commands.EditCommand.COMMAND_WORD:
```
Comment (khairulhazril) [inline src/seedu/addressbook/parser/Parser.java:82]: Try to standardize the way the cases are written

Comment (Justin-Ong) [reply src/seedu/addressbook/parser/Parser.java:82]: Yeah idk why this one refused to work until I specified the path. I wanted to just write EditCommand but it kept saying it couldn't find it.

Comment (devamanyu) [reply src/seedu/addressbook/parser/Parser.java:82]: I guess you didn't import the command. So, it was not identifying the path.

File: test/expected.txt [hunk 2]
```diff
@@ -290,6 +293,54 @@
 || 
 || 2 persons listed!
 || ===================================================
+|| Enter command: || [Command entered:  edit]
+|| Invalid command format!
```
Comment (devamanyu) [inline test/expected.txt:297]: Good job for handling i/o tests. Next, try to familiarize yourself with JUnit tests!

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -174,6 +177,59 @@ private Command prepareDelete(String args) {
...
+        if (args.split(" ").length <= 4) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, seedu.addressbook.commands.EditCommand.MESSAGE_USAGE));
+        }
+
+        args = args.trim();
+
+        int i = args.indexOf(" ");
+        String index = args.substring(0, i);
+        String pArgs = args.substring(i);
+        final int targetIndex;
```
Comment (devamanyu) [inline src/seedu/addressbook/parser/Parser.java:196]: The level of abstraction here is not consistent. Try to refactor this part of the code into separate methods to ensure better readability and coding practice.

File: src/seedu/addressbook/commands/EditCommand.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    @Override
+    public CommandResult execute() {
+        try {
+            final ReadOnlyPerson target = getTargetPerson();
+            addressBook.removePerson(target);
+            addressBook.addPerson(newPerson);
+            return new CommandResult(String.format(MESSAGE_EDIT_SUCCESS, target));
+
```
Comment (devamanyu) [inline src/seedu/addressbook/commands/EditCommand.java:65]: Stylistic advice: Try to maintain consistency on the use of whitespaces. The blank line here is probably not necessary.

**Review by devamanyu — APPROVED** [review]
Good PR to add "edit" functionality to AB-2 :+1: 

Note: Proper format to write [commit messages](https://github.com/erlang/otp/wiki/writing-good-commit-messages) : write the summary line and description of what you have done in the imperative mode, that is as if you were commanding someone. Start the line with "Fix", "Add", "Change" instead of "Fixed", "Added", "Changed".

Minor comments are provided below. Please close this PR after reading them.

**Pull request closed without merging**